drivers : flash: replace dt_nodelabel_has_prop()

kbidani opened the pull request:
…as_prop()

Replace dt_nodelabel_has_prop() with dt_compat_any_has_prop() in Kconfig.stm32_ospi

kbidani committed:
drivers : flash: replace dt_nodelabel_has_prop()

Replace dt_nodelabel_has_prop() with
dt_compat_any_has_prop() in Kconfig.stm32_ospi

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

erwango commented on drivers/flash/Kconfig.stm32_ospi:
```diff
@@ -3,8 +3,8 @@
 
-DT_STM32_OCTOSPI_1_HAS_DMA := $(dt_nodelabel_has_prop,octospi1,dmas)
-DT_STM32_OCTOSPI_2_HAS_DMA := $(dt_nodelabel_has_prop,octospi2,dmas)
+DT_STM32_OCTOSPI_1_HAS_DMA := $(dt_compat_any_has_prop,octospi1,dmas)
+DT_STM32_OCTOSPI_2_HAS_DMA := $(dt_compat_any_has_prop,octospi2,dmas)
```
Aim of `dt_compat_any_has_prop()` is to check if there is at least one node with a given compat that has a given prop.
So, it could be replaced by :
`DT_STM32_OCTOSPI_HAS_DMA := $(dt_compat_any_has_prop,<compat>,dmas)`

kbidani committed:
drivers : flash: replace dt_nodelabel_has_prop()in qspi

Replace dt_nodelabel_has_prop() with
dt_compat_any_has_prop() in Kconfig.stm32_qspi

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>

erwango commented on drivers/flash/Kconfig.stm32_qspi:
```diff
@@ -4,8 +4,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
-DT_STM32_QUADSPI_HAS_DMA := $(dt_nodelabel_has_prop,quadspi,dmas)
-
```
Please keep the line break

kbidani replied:
done.

kbidani commented:
Third commit added due to a modification at the qspi also.

gautierg-st commented on drivers/flash/Kconfig.stm32_qspi:
```diff
@@ -4,7 +4,7 @@
 # SPDX-License-Identifier: Apache-2.0
 
-DT_STM32_QUADSPI_HAS_DMA := $(dt_nodelabel_has_prop,quadspi,dmas)
+DT_STM32_QUADSPI_HAS_DMA := $(dt_compat_any_has_prop,DT_COMPAT_ST_STM32_QSPI,dmas)
```
```suggestion
DT_STM32_QUADSPI_HAS_DMA := $(dt_compat_any_has_prop,$(DT_COMPAT_ST_STM32_QSPI),dmas)
```

gautierg-st commented on drivers/flash/Kconfig.stm32_xspi:
```diff
@@ -3,8 +3,7 @@
 
-DT_STM32_XSPI_1_HAS_DMA := $(dt_nodelabel_has_prop,xspi1,dmas)
-DT_STM32_XSPI_2_HAS_DMA := $(dt_nodelabel_has_prop,xspi2,dmas)
+DT_STM32_XSPI_HAS_DMA := $(dt_compat_any_has_prop,DT_COMPAT_ST_STM32_XSPI,dmas)
```
```suggestion
DT_STM32_XSPI_HAS_DMA := $(dt_compat_any_has_prop,$(DT_COMPAT_ST_STM32_XSPI),dmas)
```

kbidani committed:
drivers : flash: replace dt_nodelabel_has_prop()in xspi

Replace dt_nodelabel_has_prop() with
dt_compat_any_has_prop() in Kconfig.stm32_xspi

Signed-off-by: Khaoula Bidani <khaoula.bidani-ext@st.com>
